### PR TITLE
Kule custom methods

### PR DESCRIPTION
**Pull request by piyapongch**
- Added Kule collection data
- Added id field on an item detail page

### CODE REVIEW

File: config/ingest.yml [hunk 1]
```diff
@@ -1,3 +1,12 @@
+# *** WARNING ***
```
Comment (murny) [inline config/ingest.yml]: Was this disclaimer and solr definition below added by mistake? Or what's this for?

File: config/SolrMarc/kule_index.properties [hunk 2]
```diff
@@ -132,7 +135,7 @@ issn_tesim = 022a:776a, first
 holdings_tesim = 949[a-z]
 owning_library_tesim = 999m
 location_tesim = 926m
-#location_tesim = "University of Alberta Internet"
+#location_tesim = "uainternet"
```
Comment (murny) [inline config/SolrMarc/kule_index.properties]: Just curious why this is commented out? If it's not needed is it safe to remove?

File: config/ingest.yml [hunk 3]
```diff
@@ -118,3 +127,8 @@ kule:
   schema: marc
   path: "data/kule.mrc"
   config: "config/SolrMarc/kule_config.properties"
+
+kule_test:
+  schema: marc
+  path: "spec/fixtures/bmufa_rev_20181010.mrc"
```
Comment (murny) [inline config/ingest.yml]: This is an bigger question for this project in general, but we really need to do a better job putting this "ingest" or "seed" data in a better location then dumping all these files in our test suite (`spec/fixtures`) especially when they are not being used for automated testing.

File: config/ingest.yml [hunk 4]
```diff
@@ -49,7 +58,7 @@ sfx_test_set:
 symphony:
   config: "config/SolrMarc/symphony_config.properties"
   schema: marc
-  path: "data/big-sample.mrc"
+  path: "data/sample.mrc"
```
Comment (pgwillia) [inline config/ingest.yml]: This needs to be 'big-sample' because that file name is hard-coded in scripts upstream.

File: .gitignore [hunk 5]
```diff
@@ -32,3 +32,5 @@ doc
 *.swp
 .env_deployment
 /docker-compose.*custom*.yml
+/.project
```
Comment (murny) [inline .gitignore]: Should probably put these changes in your personal .gitignore file and not in the project's one? More info on how to do this here:
https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: config/SolrMarc/kule_index.properties [hunk 6]
```diff
@@ -132,7 +135,7 @@ issn_tesim = 022a:776a, first
 holdings_tesim = 949[a-z]
```
Comment (weiweishi) [inline config/SolrMarc/kule_index.properties]: Also additional mapping is needed:

- Full/Alternate Title(s) field should pull from the MARC 240 $a (English, which is pulled from the 240 $l, is good)
- Published field is missing the publisher (MARC 260 $b)
- Other Physical Details field (MARC 300) is missing
- Subjects are missing subfield (MARC 650 $z, $x, $v)
- Holdings information (Call Number and Location/Status information) (MARC 926 $a, $m, $t) is missing

Comment (pojoyoho) [reply config/SolrMarc/kule_index.properties]: Can you give me all Solr fields to map?

Comment (piyapongch) [reply config/SolrMarc/kule_index.properties]: - Alternate title
Here is the mapping right now,
title_addl_t = custom, getLinkedFieldCombined(245abnps:130[a-z]:240[a-gk-s]:210ab:222ab:242abnp:243[a-gk-s]:246[a-gnp]:247[a-gnp]).
Do you want to replace all these fields with 240 $a and $l?

Comment (weiweishi) [reply config/SolrMarc/kule_index.properties]: @piyapongch you should be able to find most of the solr field mapping in the symphony mapping file. https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties 

Just a quick search found this list, and you might need to confirm. 
 
published would be:
https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties#L59
other physical details would be:
https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties#L24 mapped to 300$b
subjects should be defined here:
https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties#L52
holding information: 
call number: https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties#L67
location and status:
https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/symphony_index.properties#L145

Comment (weiweishi) [reply config/SolrMarc/kule_index.properties]: And yes, please remove the existing mapping, just use 240 $a

File: config/SolrMarc/kule_index.properties [hunk 7]
```diff
@@ -147,8 +152,16 @@ gmd_tesim = script(gmd.bsh), getGMD
...
 uniform_title_tesim = 240a
 performers_tesim = 511a
 publisher_number_tesim = 028a
+
+# Holding information
+callnumber_tesi = 926a
+status_tesi = 926t
+access_title_tesi = 999m
+access_url_tesi = "https://www.ualberta.ca/kule-folklore-centre/"
+location_tesim = 926m
```
Comment (pgwillia) [inline config/SolrMarc/kule_index.properties]: This is how symphony does this.  You'd also need to add `bmufa` and it's mapping to
https://github.com/ualbertalib/discovery/blob/master/config/SolrMarc/index_scripts/location_facet.bsh , https://github.com/ualbertalib/discovery/blob/master/config/locations.yml and https://github.com/ualbertalib/NEOSDiscovery/blob/master/config/locations.yml

```suggestion
location_tesim = script(location_facet.bsh), getLocationFacet
```

**Comment by weiweishi** [comment]
@piyapongch, Sharon has provided more test data with the new changes she'd like. Could you please test with your branch.

**Pull request closed without merging**